internal/extender/resource.go

```diff
@@ -336,92 +337,66 @@ func (s *SparkSchedulerExtender) potentialNodes(availableNodesSchedulingMetadata
 }
 
 func (s *SparkSchedulerExtender) selectExecutorNode(ctx context.Context, executor *v1.Pod, nodeNames []string) (string, string, error) {
-	resourceReservation, ok := s.resourceReservations.Get(executor.Namespace, executor.Labels[common.SparkAppIDLabel])
-	if !ok {
-		return "", failureInternal, werror.Error("failed to get resource reservations")
+	alreadyBoundNode, err := s.resourceReservationManager.FindAlreadyBoundReservationNode(ctx, executor)
+	if err != nil {
+		return "", failureInternal, werror.WrapWithContextParams(ctx, err, "error when looking for already bound reservations")
 	}
-	unboundReservations, outcome, unboundResErr := s.findUnboundReservations(ctx, executor, resourceReservation)
-	if unboundResErr != nil {
-		extraExecutorCount := 0
-		if sr, ok := s.softReservationStore.GetSoftReservation(executor.Labels[common.SparkAppIDLabel]); ok {
-			extraExecutorCount = len(sr.Reservations)
-		}
-		driver, err := s.podLister.getDriverPod(ctx, executor)
-		if err != nil {
-			return "", failureInternal, err
+	if alreadyBoundNode != "" {
```

Comment:
should we signal this with another return value from `FindAlreadyBoundReservationNode`, like it can return the `nodeName, found, error` and if `found` we can rely on a valid non empty `nodeName`

Reply:
I had done this initially, but then found it a bit redundant, although I agree it is more proper so will do this